include/alp/type_traits.hpp

```diff
@@ -344,6 +345,27 @@ namespace alp {
 			std::is_same< view::Original< void >, View >::value ||
 			std::is_same< view::Functor< typename View::applied_to >, View >::value
 		> {};
+
+		/**
+		 * Exposes the AMF type of the provided ALP container.
+		 * If a void type is passed as the ALP container type, return a generic
+		 * invalid AMF type.
+		 *
+		 * @tparam T The type to inspect.
+		 *
+		 * \note This type trait serves to enable std::conditional clauses
+		 *       that refer to members of View::applied_to, which may be void
+		 *       for containers that require allocation.
+		 */
+		template< typename T >
+		struct get_amf_type {
```

Comment:
Is this needed? Or can be encapsulated differently? At least revert the logic (default a None_type and amf_type if T is a matrix type). And if so perhaps move it close to its use?